src/cmlibs/widgets/groupmanagerwidget.py

```diff
@@ -0,0 +1,140 @@
+"""
+   Copyright 2023 University of Auckland
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+"""
+from PySide6 import QtCore, QtWidgets
+
+from cmlibs.widgets.ui.ui_groupmanagerwidget import Ui_GroupManagerWidget
+from cmlibs.utils.zinc.group import group_add_group_elements, group_remove_group_elements, \
+    group_add_not_group_elements, group_remove_not_group_elements
+
+
+NULL_PLACEHOLDER = "---"
+
+OPERATION_MAP = {
+    "Add": {"operation": group_add_group_elements, "NOT-operation": group_add_not_group_elements,
+            "tooltip": "Add all nodes/elements in this group to the selected group"},
+    "Remove": {"operation": group_remove_group_elements, "NOT-operation": group_remove_not_group_elements,
+               "tooltip": "Remove all nodes/elements in this group from the selected group"},
+}
+
+
+class GroupManagerWidget(QtWidgets.QWidget):
+    """
+    This widget takes a list of Zinc groups as an input and provides a GUI allowing the user to redefine a specific group based on the
+    nodes/elements contained in each of the other groups in the list. Calls to the GroupManagerWidget constructor must ensure that the
+    group arguments are valid and non-empty.
+
+    This widget emits a group_updated signal when the group operations are applied.
+
+    :param current_group: The FieldGroup to add elements to.
+    :param group_list: A list of FieldGroups to use for redefining the current group.
+    """
+    group_updated = QtCore.Signal()
+
+    def __init__(self, current_group, group_list, parent=None):
```

Comment:
After having a bit of a think about this I think we need to change this around.  The `__init__` method should be defined as `def __init__(self, parent=None, current_group=None, group_list=None):`.  This should allow us to use qt creator and inherited widgets.

We would also need to add API to set the current group and group list.